pkg/noun/v1/nock.c

```diff
@@ -0,0 +1,63 @@
+/// @file
+
+#include "pkg/noun/nock.h"
+#include "pkg/noun/v1/nock.h"
+
+#include "pkg/noun/v1/allocate.h"
+#include "pkg/noun/v1/hashtable.h"
+#include "pkg/noun/jets.h"
+#include "pkg/noun/v1/jets.h"
+
+/* u3n_v1_reclaim(): clear ad-hoc persistent caches to reclaim memory.
+*/
+void
+u3n_v1_reclaim(void)
+{
+  //  clear the bytecode cache
+  //
+  //    We can't just u3h_v1_free() -- the value is a post to a u3n_v1_prog.
+  //    Note that the hank cache *must* also be freed (in u3j_v1_reclaim())
+  //
+  u3n_v1_free();
+}
+
+/* _cn_prog_free(): free memory retained by program pog_u
+**              NB: copy of _cn_prog_free in pkg/noun/nock.c
+*/
+static void
+_cn_prog_free(u3n_v1_prog* pog_u)
```

Comment:
Add `_v1` for consistency.

Reply:
Copy and add the pointer recalculation logic from `_n_ream`: https://github.com/urbit/vere/blob/e98746d8856a4967eab79f716285ca71be62ed28/pkg/noun/nock.c#L2972